pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/ReadableSequentialData.java

```diff
@@ -453,12 +484,78 @@ default int readVarInt(final boolean zigZag) {
      * @throws DataEncodingException if the variable long cannot be decoded
      */
     default long readVarLong(final boolean zigZag) {
+        long pos = position();
+        long limit = limit();
+        if (limit == pos) {
+            return readVarIntLongSlow(zigZag);
+        }
+        else if (limit - pos < 10) {
+            return readVarIntLongSlow(zigZag);
+        }
+
+        long x;
+        int y;
+        if ((y = readByte()) >= 0) {
+            return zigZag ? (y >>> 1) ^ -(y & 1) : y;
+        } else if ((y ^= (readByte() << 7)) < 0) {
+            x = y ^ (~0 << 7);
+        } else if ((y ^= (readByte() << 14)) >= 0) {
+            x = y ^ ((~0 << 7) ^ (~0 << 14));
+        } else if ((y ^= (readByte() << 21)) < 0) {
+            x = y ^ ((~0 << 7) ^ (~0 << 14) ^ (~0 << 21));
+        } else if ((x = y ^ ((long) readByte() << 28)) >= 0L) {
+            x ^= (~0L << 7) ^ (~0L << 14) ^ (~0L << 21) ^ (~0L << 28);
+        } else if ((x ^= ((long) readByte() << 35)) < 0L) {
+            x ^= (~0L << 7) ^ (~0L << 14) ^ (~0L << 21) ^ (~0L << 28) ^ (~0L << 35);
+        } else if ((x ^= ((long) readByte() << 42)) >= 0L) {
+            x ^= (~0L << 7) ^ (~0L << 14) ^ (~0L << 21) ^ (~0L << 28) ^ (~0L << 35) ^ (~0L << 42);
+        } else if ((x ^= ((long) readByte() << 49)) < 0L) {
+            x ^=
+                    (~0L << 7)
+                            ^ (~0L << 14)
+                            ^ (~0L << 21)
+                            ^ (~0L << 28)
+                            ^ (~0L << 35)
+                            ^ (~0L << 42)
+                            ^ (~0L << 49);
+        } else {
+            x ^= ((long) readByte() << 56);
+            x ^=
+                    (~0L << 7)
+                            ^ (~0L << 14)
+                            ^ (~0L << 21)
+                            ^ (~0L << 28)
+                            ^ (~0L << 35)
+                            ^ (~0L << 42)
+                            ^ (~0L << 49)
+                            ^ (~0L << 56);
+            if (x < 0L) {
+                if (readByte() < 0L) {
+                    throw new DataEncodingException("Malformed VarLong");
+                }
+            }
+        }
+        return zigZag ? (x >>> 1) ^ -(x & 1) : x;
+    }
+
+    /**
+     * Read a 64bit protobuf varint at current {@link #position()}. A long var int can be 1 to 10 bytes.
+     * This is the slow path function.
+     *
+     * @return long read in var int format
+     * @param zigZag use protobuf zigZag varint encoding, optimized for negative numbers
+     * @throws BufferUnderflowException If the end of the sequence is reached before the final variable byte fragment
+     *                                  is read
+     * @throws DataAccessException if an I/O error occurs
+     * @throws DataEncodingException if the variable long cannot be decoded
+     */
+    default long readVarIntLongSlow(final boolean zigZag) {
```

Comment:
Why are we adding readVarIntLongSlow() to api? It was only interesting in testing should not be committed to API